tests/test_endl.py

```diff
@@ -0,0 +1,58 @@
+import os
+import io
+import warnings
+try:
+    from StringIO import StringIO
+except ImportError:
```

Comment:
I am assuming that this is because of Python 2/3 compatibility.  In this case you should use something like the following.

```python
import sys
if sys.version_info[0] > 2:
   from io import StringIO
else:
   from StringIO import StringIO
```